test/test_pass_fds.rs

```diff
@@ -0,0 +1,37 @@
+extern crate nix;
+
+use std::os::unix::net::UnixDatagram;
+use std::os::unix::io::{RawFd, AsRawFd};
+use nix::sys::socket::{CmsgSpace, ControlMessage, MsgFlags, sendmsg, recvmsg};
+use nix::sys::uio::IoVec;
+
+
+fn main() {
+    let (send, receive) = UnixDatagram::pair().unwrap();
+    let thread = std::thread::spawn(move || {
+        let mut buf = [0u8; 8];
+        let iovec = [IoVec::from_mut_slice(&mut buf)];
+        let mut space = CmsgSpace::<[RawFd; 2]>::new();
+        let (paneid, fds) = match recvmsg(receive.as_raw_fd(), &iovec, Some(&mut space), MsgFlags::empty()) {
+            Ok(msg) => {
+                let mut iter = msg.cmsgs();
+                if let Some(ControlMessage::ScmRights(fds)) = iter.next() {
+                    (iovec[0].as_slice(), fds.to_vec())
+                } else {
+                    panic!();
+                }
+            },
+            Err(_) => {
+                panic!();
```

Comment:
These two panic messages are indistinguishable since you don't provide descriptions for them.  And if the error case is a simple `panic`, you may as well just use `unwrap` instead of `if let`.